File: java/3_operator/bitwise/Not.java

class Not {
    
    public static void main(String[] args){
        int a = 3; String bitA;
        int hasil = ~a;
        String bitHasil;
        
        bitA = String.format("%8s", Integer.toBinaryString(a)).replace(' ', '0');
        bitHasil = String.format("%8s", Integer.toBinaryString(hasil)).replace(' ','0');

        bitHasil = bitHasil.substring(bitHasil.length() -8);
        System.out.println("Hasil bitwise not variable a adalah:");
        System.out.println(bitA);
        System.out.println(bitHasil);
    }   
}

File: kotlin/3_operator/bitwise/Not.kt

fun main() {

    val a: Int = 3
    var hasil = a.inv()

    val bitA = a.toString(2).padStart(8,'0')
    val bitHasil = hasil.toString(2).padStart(8,'0')

    println("Hasil bitwise not variable a adalah:")
    println(bitA)
    println(bitHasil)
}
